src/dune_rules/melange/melange_stanzas.ml

```diff
@@ -29,25 +29,48 @@ module Emit = struct
     let add kind name acc =
       match Lib_name.Map.find acc name with
       | None -> Lib_name.Map.set acc name kind
-      | Some _present ->
-        User_error.raise ~loc
-          [ Pp.textf "library %S is present twice" (Lib_name.to_string name) ]
+      | Some kind' -> (
+        match (kind, kind') with
```

Comment:
Seems like this is the same check as in dune_file.ml for normal libraries. Can we share the check?

Reply:
did that in [917bb1a](https://github.com/ocaml/dune/pull/7239/commits/917bb1ae8c5d88d61d9a3bdede7f27e48e512558), ended up using the whole decoder.